docs/design/bundler.md

```diff
@@ -0,0 +1,516 @@
+"#{base_name}-#{shortref_for_path(revision)}"
+```
+
+The name of the directory **must come from the Git URL**, not the actual name of the gem, and the cloned folder must
+contain unpacked source code. Any other format will cause bundler to try to re-download the repository, causing the
+build to fail.
+
+##### Multiple Gems in a single repository
+
+A single repository can hold multiple Gems, and those can be imported as dependencies. When this happens, Bundler still
+expects a single clone to be made. Here's an example of how multiple gems imported from a single repository+revision
+looks like in the `Gemfile.lock`:
+
+```
+GIT
+  remote: https://github.com/chatwoot/azure-storage-ruby
+  revision: 9957cf899d33a285b5dfe15bdb875292398e392b
+  branch: chatwoot
+  specs:
+    azure-storage-blob (2.0.3)
+      azure-storage-common (~> 2.0)
+      nokogiri (~> 1, >= 1.10.8)
+    azure-storage-common (2.0.4)
+      faraday (~> 2.0)
+      faraday-follow_redirects (~> 0.3.0)
+      faraday-net_http_persistent (~> 2.0)
+      net-http-persistent (~> 4.0)
+      nokogiri (~> 1, >= 1.10.8)
+```
+
+### Out of scope
+
+#### Plugins
+Bundler has support for using [plugins](https://bundler.io/guides/bundler_plugins.html), which allows users to extend
+Bundler's functionality in any way that they seem fit. Since this can open the possibility for security issues, plugins
+will not be supported by Cachi2.
+
+Since we're not proposing the direct usage of Bundler to fetch the dependencies, no other actions are needed in the
+prefetch phase, existing plugin definitions will be ignored.
+
+#### Pre-compiled binaries
+For the initial implementation, we're aiming to provide support only for plain Ruby gems (which are idenfied as `ruby`
+in the `PLATFORMS` section of the `Gemfile.lock`). Platforms that relate to specific architectures will contain
+binaries that were pre-compiled for that architecture (see [Platforms](#platforms)).
+
+The URL schema in the default rubygems registry seems to follow this format:
+
+```ruby
+# Plain Ruby Gem
+"https://rubygems.org/gems/#{name}-#{version}.gem"
+
+# Platform-specific Gem
+"https://rubygems.org/gems/#{name}-#{version}-#{platform}.gem"
+```
+
+This means that we can easily just download the plain Ruby Gems, and skip all platform-specific Gems altogether. From
+the testing I did, this did not seem to affect the ability to perform a hermetic build in any way, Bundler just
+compiled everything from source.
+
+To avoid confusing users if a corner case shows up, we can add a `WARNING` log in case extra platforms are detected,
+and also document it properly.
+
+Proper support for pre-compiled binaries should be probably left as a follow-up feature, similarly to what was done
+with [pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
```

Comment:
typo: s/`Gemlock.file`/`Gemfile.lock` file

Reply:
I'll blame that one on Javascript (and it's package-lock file) :smile: